### PR TITLE
Add frame filtering feature

### LINKED ISSUE #537: Frame filtering feature
When conditional video encoding is used, future pipelines can receive frames without video data. We need a feature to handle them and avoid passing them to the video pipeline.

Like:

```
ingress_frame_filter:
  module: module (in the unified YAML format)
  class_name: Class

egress_frame_filter:
  module: module (in the unified YAML format)
  class_name: Class
```

works with VideoFrame objects (savant-rs)

- True: pass
- False: drop

By default, ingress filters out all frames without video, and egress is bypassed (`return True`). 
Works right after zmq source and before zmq sink.

### PR DESCRIPTION
**Pull request by abramov-oleg**

### CODE REVIEW

File: .devcontainer/l4t/devcontainer.json [hunk 1]
```diff
@@ -14,7 +15,7 @@
   "runArgs": [ "--runtime=nvidia" ],
   "containerEnv": {
     "LOGLEVEL": "DEBUG",
-    "ZMQ_SRC_ENDPOINT": "sub+bind:ipc:///tmp/zmq-sockets/input-video.ipc",
+    "ZMQ_SRC_ENDPOINT": "router+bind:ipc:///tmp/zmq-sockets/input-video.ipc",
```
Comment (tomskikh) [inline .devcontainer/l4t/devcontainer.json:18]: Why this was changed?

File: savant/config/module_config.py [hunk 2]
```diff
@@ -307,12 +347,21 @@ def configure_pipeline_elements(module_cfg: DictConfig) -> None:
         module_cfg.pipeline = OmegaConf.structured(Pipeline)
         return
 
+    logger.debug('Configure pipeline source...')
```
Comment (tomskikh) [inline savant/config/module_config.py]:
```suggestion
    logger.debug('Configuring pipeline source...')
```

Same for https://github.com/insight-platform/Savant/pull/573/files#diff-0e0d7e987cb17ddd4aff5ffe31878d15fdc23cc58e6d4826894bab5c1708c5e4R355

File: savant/config/schema.py [hunk 3]
```diff
@@ -328,6 +328,43 @@ def get_element_name(element: Union[DictConfig, PipelineElement]) -> str:
...
+class SourceElement(PipelineElement):
+    """A pipeline element that produces pipeline input."""
+
+    ingress_frame_filter: PyFunc = PyFunc(
+        module='savant.base.frame_filter', class_name='DefaultIngressFilter'
+    )
+    """Frame filter for ingress frames."""
+
+    def __post_init__(self):
+        if self.element == 'zeromq_source_bin':
```
Comment (tomskikh) [inline savant/config/schema.py:347]: Log warning or stop the module with an error when `ingress_frame_filter` is configured for source that is not `zeromq_source_bin`.

File: savant/utils/sink_factories.py [hunk 4]
```diff
@@ -59,6 +61,54 @@ def source_id(self) -> str:
...
+            logger.warning(
+                'Error in sink "%s" loading egress filter %s: %s. '
+                'Using noop placeholder.',
+                self.name,
+                self.egress_pyfunc,
+                exc,
+            )
+            self.egress_pyfunc = lambda x: True
+
+    def video_frame_filt(self, video_frame: VideoFrame, frame_pts: int) -> bool:
```
Comment (tomskikh) [inline savant/utils/sink_factories.py]: Why do you need a `frame_pts` here?

Comment (tomskikh) [inline savant/utils/sink_factories.py]: `video_frame_filter`

File: gst_plugins/python/savant_rs_video_demux.py [hunk 5]
```diff
@@ -114,27 +115,40 @@
         False,
         GObject.ParamFlags.READWRITE,
     ),
+    'ingress-module': (
+        str,
+        'Python module',
+        'Python module name to import or module path.',
```
Comment (tomskikh) [inline gst_plugins/python/savant_rs_video_demux.py]: Describe what is this python module. And for other properties too.

File: gst_plugins/python/savant_rs_video_demux.py [hunk 6]
```diff
@@ -222,6 +245,24 @@ def do_state_changed(self, old: Gst.State, new: Gst.State, pending: Gst.State):
...
+                    self.ingress_kwargs,
+                    self.ingress_dev_mode,
+                )
+            else:
+                # for AO RTSP
+                self.logger.debug(
+                    'Ingress filter is not set, '
+                    'using default check for video frame content.'
+                )
+                self.ingress_pyfunc = lambda x: not x.content.is_none()
```
Comment (tomskikh) [inline gst_plugins/python/savant_rs_video_demux.py]: Use default filter explicitly (`self.ingress_pyfunc = DefaultInressFilter()`).

**Review by tomskikh — CHANGES_REQUESTED** [review]

File: .devcontainer/l4t/devcontainer.json (hunk 1 above)
Comment (abramov-oleg) [reply .devcontainer/l4t/devcontainer.json:18]: In order to bring it in line with `socket_type` in the `savant/config/default.yml` and the default value for `--socket` option in `scripts/uri-input.py`.

File: savant/utils/sink_factories.py (hunk 4 above)
Comment (abramov-oleg) [reply savant/utils/sink_factories.py]: Renamed.

File: gst_plugins/python/savant_rs_video_demux.py (hunk 6 above)
Comment (abramov-oleg) [reply gst_plugins/python/savant_rs_video_demux.py]: Used the `DefaultInressFilter` class.

File: gst_plugins/python/savant_rs_video_demux.py (hunk 5 above)
Comment (abramov-oleg) [reply gst_plugins/python/savant_rs_video_demux.py]: Improved the description.

File: savant/utils/sink_factories.py (hunk 4 above)
Comment (abramov-oleg) [reply savant/utils/sink_factories.py]: In order to make error log messages have the same format as zeromq sink debug messages.

File: savant/config/schema.py (hunk 3 above)
Comment (abramov-oleg) [reply savant/config/schema.py:347]: Made `ingress_frame_filter` optional for SourceElement and added a warning for the case when it is configured for sources other than `zeromq_source_bin`.

Updated docs.

File: savant/utils/sink_factories.py (hunk 4 above)
Comment (tomskikh) [reply savant/utils/sink_factories.py]: Ok. Move `frame_pts = convert_ts(...)` inside `video_frame_filter`, there's no need to convert the timestamp before each call of the function.

**Review by tomskikh — APPROVED** [review]

**Pull request merged**